sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -370,17 +365,20 @@ def clear(self):
     self._added_elements = []
 
   def _commit(self):
+    to_await = None
     if self._cleared:
-      self._state_handler.blocking_clear(self._state_key)
+      to_await = self._state_handler.clear(self._state_key, is_cached=True)
     if self._added_elements:
-      value_coder_impl = self._value_coder.get_impl()
-      out = coder_impl.create_OutputStream()
-      for element in self._added_elements:
-        value_coder_impl.encode_to_stream(element, out, True)
-      self._state_handler.blocking_append(self._state_key, out.get())
+      to_await = self._state_handler.append(
```

Comment:
Just a thought: I wonder if append should be renamed extend (analogous to list.append vs list.extend) as we're concatenating multiple elements here now.

Reply:
Good suggestion, although I think we have been adding multiple elements before.

Reply:
Done.